numpy/lib/tests/test_function_base.py

```diff
@@ -1597,6 +1597,12 @@ def test_empty(self):
             assert_array_equal(corrcoef(np.array([]).reshape(2, 0)),
                                np.array([[np.nan, np.nan], [np.nan, np.nan]]))
 
+    def test_extreme(self):
+        x = [[1e-100, 1e100], [1e100, 1e-100]]
+        with np.errstate(all='raise'):
+            c = corrcoef(x)
+        assert_array_equal(c, np.array([[1., -1.], [-1., 1.]]))
```

Comment:
Maybe `assert_array_almost_equal` with high accuracy would be safer?  None of the numbers are exactly representable in binary, so there is bound to be a bit or roundoff error that may be compiler/platform dependent.
```
In [1]: print "%27.20e" % 1e100
1.00000000000000001590e+100
```